.github/workflows/docker-publish.yaml

```diff
@@ -0,0 +1,134 @@
+
+      - name: Create revision file
+        run: echo $(date +%s)-${{ steps.get-commit-id.outputs.commit }} >./revision.txt
+
+      - id: get-version
+        name: Get version
+        run: echo "version=${{ startsWith(github.ref, 'refs/tags/') && github.ref_name || '' }}" >> $GITHUB_OUTPUT
+
+      - name: Set package version
+        if: ${{ steps.get-version.outputs.version }} # ${GITHUB_REF} does not have a tag if this is not set above
+        run: npm version ${GITHUB_REF#refs/tags/v} --no-git-tag-version --allow-same-version
+
+      - id: set-environment
+        name: Set environment
+        run: echo "environment=${{ startsWith(github.ref, 'refs/tags/') && 'prod' || 'dev' }}" >> $GITHUB_OUTPUT
+
+      - name: Install all dependencies
+        run: npm ci --ignore-scripts
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+
+      - name: Run post install scripts of dependencies
+        run: npm rebuild && npm run prepare --if-present
+
+      - name: Build
+        run: npm run build
+
+      - name: Install dependencies without devDependencies
+        run: npm ci --omit=dev --ignore-scripts
+        env:
+          NODE_AUTH_TOKEN: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+
+      - name: Run post install scripts of dependencies
+        run: npm rebuild && npm run prepare --if-present
+
+      - name: Build and push Docker image to AWS ECR
+        run: |
+          REPO="${{ vars.AWS_ACCOUNT_ID || '915497967985' }}.dkr.ecr.${{ vars.AWS_REGION }}.amazonaws.com"
+          IMAGE="$REPO/microsoft-teams-app:${{ startsWith(github.ref, 'refs/tags/') && github.ref_name || steps.get-commit-id.outputs.commit }}"
+          IMAGE_WITH_ENV_TAG="$REPO/microsoft-teams-app:${{ steps.set-environment.outputs.environment }}"
+          docker build -t $IMAGE .
+          docker tag $IMAGE $IMAGE_WITH_ENV_TAG
+          aws ecr get-login-password --region ${{ vars.AWS_REGION }} | docker login --username AWS --password-stdin $REPO
+          docker push $IMAGE
+          docker push $IMAGE_WITH_ENV_TAG
+
+      - name: Trigger infra workflow to deploy new image version
+        run: |
+          JSON_DATA=$(
+            jq -n -c \
+              --arg b "${{ github.ref_name }}" \
+              --arg t "${{ github.ref_name }}" \
+              --arg v "${{ steps.get-version.outputs.version }}" \
+              --arg c "${{ steps.get-commit-id.outputs.commit }}" \
+              --arg e "${{ steps.set-environment.outputs.environment }}" \
+            '{"app-name": "microsoft-teams-app", "branch": $b, "version": $v, "tag": $t, "commit-id": $c, "environment": $e }'
+          )
+          echo $JSON_DATA | gh workflow run deploy-app.yaml --repo zeplin/infra --json
+          echo "Workflow is triggered: https://github.com/zeplin/infra/actions/workflows/deploy-app.yaml"
+        env:
+          GH_TOKEN: ${{ secrets.GH_PERSONAL_ACCESS_TOKEN }}
+
+  post-deploy:
```

Comment:
As far as I see, this is similar to scim. Can you remind me why we need these in this file? I mean this could be a dev publish. (@yuqu)

Reply:
If this is a dev publsih it will just create release markers on Newrelic and Sentry for dev environment, we won't see it for prod environment

Reply:
Newrelic shows "related changes" as markers, I'd definitely not want to see it as a related change in its prod version, for example.

Also, the reason I was confused was because I forgot we'll get rid off `development.yaml` and `production.yaml` files. We should check if the new docker publish does everything in those files in other services, too (for example public-api does not put markets in the docker publish file).